homeassistant/components/litterrobot/binary_sensor.py

```diff
@@ -0,0 +1,95 @@
+"""Support for Litter-Robot binary sensors."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Generic
+
+from pylitterbot import LitterRobot, Robot
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .entity import LitterRobotEntity, _RobotT
+from .hub import LitterRobotHub
+
+
+@dataclass
+class RequiredKeysMixin(Generic[_RobotT]):
+    """A class that describes robot binary sensor entity required keys."""
+
+    is_on_fn: Callable[[_RobotT], bool]
+
+
+@dataclass
+class RobotBinarySensorEntityDescription(
+    BinarySensorEntityDescription, RequiredKeysMixin[_RobotT]
+):
+    """A class that describes robot binary sensor entities."""
+
+
+class LitterRobotBinarySensorEntity(LitterRobotEntity[_RobotT], BinarySensorEntity):
+    """Litter-Robot binary sensor entity."""
+
+    entity_description: RobotBinarySensorEntityDescription[_RobotT]
+
+    @property
+    def is_on(self) -> bool:
+        """Return the state."""
+        return self.entity_description.is_on_fn(self.robot)
+
+
+BINARY_SENSOR_MAP: dict[type[Robot], tuple[RobotBinarySensorEntityDescription, ...]] = {
+    LitterRobot: (
+        RobotBinarySensorEntityDescription[LitterRobot](
+            key="is_sleeping",
+            name="Is sleeping",
```

Comment:
Would Sleeping be better?

Reply:
I think that's more appropriate here as well now